alexandria/storages/fields.py

```diff
@@ -0,0 +1,42 @@
+from django.conf import ImproperlyConfigured, settings
+from django.db import models
+from django.db.models.fields.files import FieldFile
+
+from alexandria.storages.backends.s3 import S3Storage, SsecGlobalS3Storage
+
+
+class DynamicStorageFieldFile(FieldFile):
+    def __init__(self, instance, field, name):
+        super().__init__(instance, field, name)
+        if (
+            settings.ALEXANDRIA_ENABLE_AT_REST_ENCRYPTION
+            and instance.encryption_status == instance.EncryptionStatus.SSEC_GLOBAL_KEY
+        ):
+            self.storage = SsecGlobalS3Storage()
+
+
+class DynamicStorageFileField(models.FileField):
+    attr_class = DynamicStorageFieldFile
+
+    def pre_save(self, instance, add):
+        if settings.ALEXANDRIA_ENABLE_AT_REST_ENCRYPTION:
+            if (
+                method := settings.ALEXANDRIA_ENCRYPTION_METHOD
+            ) not in instance.EncryptionStatus.values:
+                msg = (
+                    f"ALEXANDRIA_ENCRYPTION_METHOD must be one of "
+                    f"{instance.EncryptionStatus.values}. {method} is not valid"
+                )
+                raise ImproperlyConfigured(msg)
+            if not (storage := SsecGlobalS3Storage()) and isinstance(
+                storage, S3Storage
+            ):
```

Comment:
:seal:

Reply:
that anyway did nothing useful. The check should have been with the default backend to avoid another layer of complexity with potentially different backend-types per file.